File: code/jvm/src/main/kotlin/pt/isel/leic/ptgest/http/controllers/workout/model/request/CreateCustomExerciseRequest.kt
package pt.isel.leic.ptgest.http.controllers.workout.model.request

import jakarta.validation.constraints.NotEmpty
import jakarta.validation.constraints.Size
import pt.isel.leic.ptgest.domain.workout.Modality
import pt.isel.leic.ptgest.domain.workout.MuscleGroup

data class CreateCustomExerciseRequest(
    @field:NotEmpty(message = "Name cannot be empty")
    val name: String,
    val description: String?,
    @field:Size(max = 3, message = "Muscle group list cannot have more than 3 elements")
    val muscleGroup: List<MuscleGroup>,
    val modality: Modality,
    val ref: String?
) {
    init {
        name.trim()
        description?.trim()
        ref?.trim()
    }
}

File: code/js/src/views/user/TrainerViews/models/CreateCustomExerciseRequest.ts
class CreateCustomExerciseRequest{
    name: string
    description: string | null
    muscleGroup: string[]
    modality: string
    ref: string| null

    constructor(name: string, modality: string , muscleGroup: string[],description: string | null, ref: string | null) {
        this.name = name;
        this.modality = modality;
        this.muscleGroup = muscleGroup;
        this.description = description;
        this.ref = ref;
    }
}

export default CreateCustomExerciseRequest;

